pkg/controller/appdefinition/deploy.go

```diff
@@ -234,7 +234,50 @@ func toContainers(app *v1.AppInstance, tag name.Reference, name string, containe
 		})
 	}
 
+	if addBusybox {
+		// Drop the static busybox binary into a shared volume so that we can use it in initContainers.
+		initContainers = append(initContainers, corev1.Container{
+			Name:            "acorn-helper-busybox",
+			Image:           system.DefaultImage(),
+			Command:         []string{"acorn-busybox-init"},
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      sanitizeVolumeName(AcornHelper),
+					MountPath: AcornHelperPath,
+				},
+			},
+		},
+		)
+	}
+
 	newContainer := toContainer(app, tag, name, container, interpolator, addWait && len(container.Ports) > 0)
+	for src, dir := range container.Dirs {
+		if dir.Preload {
+			// If a directory is marked for preload, then add an initContainer that copies the directory into the shared volume.
+			// Data will be copied to the data/ subdirectory and we'll drop a .preload-done file in the root to indicate that
+			// the copy has been completed (and should not be repeated).
+			initContainers = append(initContainers, corev1.Container{
+				Name:            "acorn-preload-dir-" + sanitizeVolumeName(src),
```

Comment:
It is possible, though unlikely, for a user to define an init container with this name. Should we add some randomness to it?

Reply:
I'll just add the app UID to it :)
It's already unlikely enough, given that we're using the path hash, so this should do it.